### PR TITLE
Add a new rz_buf_write_* API

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

`rz_buf_write_ble*`, `rz_buf_write_le*`, `rz_buf_write_be*` functions were added to simplified relocations patching.

**Feedback needed**:
* There is a lot of code repetition, so maybe the use of macro to generate `rz_buf_write*` and `rz_buf_read*` functions could be used (but it is harder to document).
* I don't know If I should use macro to instead of function definition for `rz_buf_write_le*` and `rz_buf_write_be*`.
* Should `rz_buf_write_*` and `rz_buf_read_*` be inline functions?

Example:
see https://github.com/rizinorg/rizin/blob/f076be829ab69593473dd9c67cf548b054fd8d03/librz/include/rz_util/rz_num.h#L124

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by ret2libc** [comment]
> * There is a lot of code repetition, so maybe the use of macro to generate `rz_buf_write*` and `rz_buf_read*` functions could be used (but it is harder to document).

> * I don't know If I should use macro to instead of function definition for `rz_buf_write_le*` and `rz_buf_write_be*`.
> * Should `rz_buf_write_*` and `rz_buf_read_*` be inline functions?

I think they can be inline functions, yes. And probably they can be defined in the header through macros to avoid the dupped code. Smart editors should be able to reference them anyway and yes, with good documentation they should be understandable.

**Comment by ghost** [comment]
> > * There is a lot of code repetition, so maybe the use of macro to generate `rz_buf_write*` and `rz_buf_read*` functions could be used (but it is harder to document).
> 
> > * I don't know If I should use macro to instead of function definition for `rz_buf_write_le*` and `rz_buf_write_be*`.
> > * Should `rz_buf_write_*` and `rz_buf_read_*` be inline functions?
> 
> I think they can be inline functions, yes. And probably they can be defined in the header through macros to avoid the dupped code. Smart editors should be able to reference them anyway and yes, with good documentation they should be understandable.

Okay, I will try tomorrow

**Comment by ghost** [comment]
I have updated the PR, what do you think? ~And there is some warning during the compilation about undefined variable. I don't have any idea why.~

PS: Some documentation should be added. If you have any suggestion?

File: librz/include/rz_util/rz_buf.h [hunk 1]
```diff
@@ -157,6 +141,107 @@ static inline st64 rz_buf_sleb128_at(RzBuffer *b, ut64 addr, st64 *v) {
...
+ * \param result ...
+ * \param big_endian ...
+ * \return Return the status of the operation.
+ */
+DEFINE_RZ_BUF_READ_BLE(16)
+DEFINE_RZ_BUF_READ_BLE(32)
+DEFINE_RZ_BUF_READ_BLE(64)
+
+/**
+ * \brief Read a big endian or little endian ut16 at the specified address or cursor in the buffer.
```
Comment (XVilka) [inline librz/include/rz_util/rz_buf.h:199]: @ret2libc do you know if it works?

File: librz/bin/p/bin_ne.c [hunk 2]
```diff
@@ -16,6 +16,7 @@ static bool check_buffer(RzBuffer *b) {
 
 	ut16 idx;
 	if (!rz_buf_read_le16_at(b, 0x3c, &idx)) {
+		return false;
 	}
```
Comment (XVilka) [inline librz/bin/p/bin_ne.c]: @08A please send this fix as a separate PR, we will include it in 0.3.0

Comment (ghost) [reply librz/bin/p/bin_ne.c]: A new PR was opened.

File: librz/include/rz_util/rz_buf.h (hunk 1 above)
Comment (ret2libc) [reply librz/include/rz_util/rz_buf.h:199]: I think it can be made to work with some magic from https://www.doxygen.nl/manual/preprocessing.html

**Review by XVilka — APPROVED** [review]

**Comment by ghost** [comment]
The clang format CI failed because the llvm download failed.

**Review by wargio — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]
